cranelift-codegen/src/legalizer/boundary.rs

```diff
@@ -644,10 +870,39 @@ pub fn handle_return_abi(inst: Inst, func: &mut Function, cfg: &ControlFlowGraph
                 .ebb_params(pos.func.layout.entry_block().unwrap())[idx];
             debug_assert_eq!(pos.func.dfg.value_type(val), arg.value_type);
             vlist.push(val, &mut pos.func.dfg.value_lists);
+
+            if let ArgumentPurpose::StructReturn = arg.purpose {
+                sret = Some(val);
```

Comment:
Unless I am overlooking something, I don't think there is an easy way to do it in the `match` above, because we don't have the `val` inserted in the value list at that point, and we would have to duplicate almost all the rest of this loop's body.

Reply:
You're right, sorry for the noise.